src/PhpSlang/Collection/Generic/Collection.php

```diff
@@ -5,12 +5,14 @@
 namespace PhpSlang\Collection\Generic;
 
 use Closure;
+use Countable;
+use Iterator;
 use PhpSlang\Collection\HashMapCollection;
 use PhpSlang\Collection\ListCollection;
 use PhpSlang\Collection\SetCollection;
 use PhpSlang\Option\Option;
 
-interface Collection
+interface Collection extends Iterator, Countable
```

Comment:
where is the `Countable::count()` method implemented?

Reply:
btw this is a compatibility break, I guess the PR should have the target branch set to some 2.0-dev created from master